### PR TITLE
Feat: create deep link page for messages 

### LINKED ISSUE #2020: [EIP1271] Show shareable links in MsgModal and history
Use this endpoint (routed via CGW) to fetch message details:
https://safe-transaction-goerli.safe.global/api/v1/messages/0x7f4032fd13f27c5fce4694a1a6b79f65c68656da4eee4347a414f5bcec915b39/

Create a deep link page for Message details.

Link to it from the Message popup and History.
Screenshot 2023-05-22 at 13 40 25

## Designs
[screen view](https://www.figma.com/file/cMDccuR9aZFqAiBd4dKsKg/Transaction-flow?type=design&node-id=1%3A15710&mode=design&t=FLMcU43UXdbatjOb-1)
[states](https://www.figma.com/file/cMDccuR9aZFqAiBd4dKsKg/Transaction-flow?type=design&node-id=1%3A16071&mode=design&t=FLMcU43UXdbatjOb-1)

### PR DESCRIPTION
**Pull request by jmealy**
## What it solves

Resolves #2020

## How this PR fixes it
- Create a deep link page for displaying a single message.
- Link to this single message page from the sign message modal and from the message details in the history.

## How to test it
- Create a message for a 2/n safe (eg. log in to opensea via walletconnect)
- Sign the message using the first signer of the safe.
- Once the message has one signature, there should be a link to share the message with the other signers [1]
- Go to the message history. There should also be a share link in the message details, similar to transactions. [2]
- The link should lead to a page displaying that message by itself.

## Screenshots
[1]
image
[2]
image
## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/components/safe-messages/SingleMsg/SingleMsg.test.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import * as syncSafeMessageSigner from '@/hooks/messages/useSyncSafeMessageSigner'
+
+import { SafeMessageListItemType, SafeMessageStatus, type SafeMessage } from '@safe-global/safe-gateway-typescript-sdk'
+import SingleMsg from '.'
+
+const SAFE_ADDRESS = '0x87a57cBf742CC1Fc702D0E9BF595b1E056693e2f'
+const SAFE_MESSAGE_HASH = '0x32d898a35478c1c2063f76841cd99388e28ddac217e17bf787e2307ba8bf860b'
+
+// Minimum mock to render <SingleMsg />
+const msgDetails = {
```
Comment (schmanu) [inline src/components/safe-messages/SingleMsg/SingleMsg.test.tsx:13]: Nice to have:
Ideally we create a `safeMsgBuilder` for this. Similar to our `safeInfoBuilder` and `safeTxBuilder`. Really handy to mock quickly.

File: src/components/safe-messages/SingleMsg/index.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import ExpandableMsgItem from '../MsgListItem/ExpandableMsgItem'
+import ErrorMessage from '@/components/tx/ErrorMessage'
+
+const SingleMsg = () => {
+  const router = useRouter()
+  const { messageHash } = router.query
+  const safeMessageHash = Array.isArray(messageHash) ? messageHash[0] : messageHash
+  const { safe, safeAddress } = useSafeInfo()
+
+  const [message, messageError] = useAsync<SafeMessage>(
```
Comment (schmanu) [inline src/components/safe-messages/SingleMsg/index.tsx:17]: There is already a `useSafeMessage` hook that fetches a message by hash

**Review by schmanu — APPROVED** [review]
Just some minor comments. Otherwise it looks good so I'll already approve.

File: src/components/tx-flow/flows/SignMessage/SignMessage.tsx [hunk 3]
```diff
@@ -322,14 +324,14 @@ const SignMessage = ({ message, safeAppId, requestId }: ProposeProps | ConfirmPr
...
+            <AlreadySignedByOwnerMessage hasSigned={signedByCurrentSafe} />
 
             <InfoBox
               title="Collect all the confirmations"
               message={
-                requestId
+                requestId && !hasSignature
                   ? 'Please keep this modal open until all signers confirm this message. Closing the modal will abort the signing request.'
-                  : 'The signature will be submitted to the Safe App when the message is fully signed.'
+                  : 'The signature will be submitted to the App when the message is fully signed.'
```
Comment (katspaugh) [inline src/components/tx-flow/flows/SignMessage/SignMessage.tsx:334]:
```suggestion
                  : 'The signature will be submitted to the requesting app when the message is fully signed.'
```

Technically, `requestId` is there only for Safe App messages. Swaps also, but they don't have messages (yet).

**Comment by francovenica** [comment]
LGTM

The link is fine, it copies the the url properly. 
Looks good in dark mode and mobile viewports

**Pull request merged**